appsec/src/helper/remote_config/listeners/asm_features_listener.cpp

```diff
@@ -36,18 +67,31 @@ void dds::remote_config::asm_features_listener::on_update(const config &config)
         if (dd_tolower(enabled_itr->value.GetString()) == std::string("true")) {
             service_config_->enable_asm();
         } else {
-            // This scenario should not happen since RC would remove the file
-            // when appsec should not be enabled
             service_config_->disable_asm();
         }
     } else if (enabled_itr->value.GetType() == rapidjson::kTrueType) {
         service_config_->enable_asm();
     } else if (enabled_itr->value.GetType() == rapidjson::kFalseType) {
-        // This scenario should not happen since RC would remove the file
-        // when appsec should not be enabled
         service_config_->disable_asm();
     } else {
         throw error_applying_config(
             "Invalid config json encoded contents: enabled key invalid");
     }
 }
+
+void dds::remote_config::asm_features_listener::on_update(const config &config)
+{
+    rapidjson::Document serialized_doc;
+    if (!json_helper::get_json_base64_encoded_content(
+            config.contents, serialized_doc)) {
+        throw error_applying_config("Invalid config contents");
+    }
+
+    if (dynamic_enablement_) {
+        parse_asm(serialized_doc);
+    }
+
+    if (api_security_enabled_) {
+        parse_api_security(serialized_doc);
+    }
+}
```

Comment:
I think it would make more sense to evaluate all configurations and then apply at the end, I would also change the parser to ensure that if, in the future, we add more overlapping configurations (e.g. a configuration with the `asm` key but not containing `enabled`) that existing libraries don't suddenly report errors.

Reply:
I would leave it as it is. Right now this listener is in charge of two different and independent things. The methods are actually doing what you say, first parse and then apply.
Regarding relaxing errors, I could but that would be risky. As of now, when this configuration comes with `asm` key, it needs to be an object and it needs to have the `enable` key. If we can't rely on that, we would enter on a paranoiac  approach. Also as of know if configurations don't get apply, we won't know as there are no errors.